eden/scm/lib/gitcompat/src/config.rs

```diff
@@ -22,12 +22,17 @@ impl GlobalGit {
                 &["--show-scope", "--get-regexp", "^(remote|user)\\."],
             )
             .output()?;
-        let out = String::from_utf8(out.stdout)?;
-        Ok(translate_git_config_output(&out))
+        let config = String::from_utf8(out.stdout)?;
+        let remotes_out = self
+            .git_cmd(
+                "ls-remote", &["--symref", ".", "HEAD"])
```

Comment:
This performs a network call, which might be unexpected or undesirable.

How about parsing the output of `git rev-parse --symbolic refs/remotes/origin/HEAD` instead?

Reply:
Good idea!  If it's as robust as `ls-remote` then this will save a network call.

Reply:
You can use [`lookup_reference`](https://github.com/facebook/sapling/blob/a244654809aef840ef06c8dd4d08acf2ce1ac6ba/eden/scm/lib/gitcompat/src/refs.rs#L56) to check the reference without shelling out.